src/main/java/uk/gov/companieshouse/siccode/api/search/SicCodeService.java

```diff
@@ -0,0 +1,34 @@
+package uk.gov.companieshouse.siccode.api.search;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.query.TextCriteria;
+import org.springframework.stereotype.Service;
+
+@Service
+public class SicCodeService {
+
+    @Autowired
+    private CombinedSicActivitiesRepository combinedSicActivitiesRepository;
+
+    public  List<CombinedSicActivitiesStorageModel> search(SicCodeSearchRequestApiModel sicCodeSearchRequestApiModel) {
+
+        if (StringUtils.isEmpty(sicCodeSearchRequestApiModel.getSearchString())) {
+            return new ArrayList<>();
+        }
+
+        TextCriteria criteria = null;
+        if (sicCodeSearchRequestApiModel.isMatchPhrase()) {
+            criteria = new SicCodeSearchTextCriteria(sicCodeSearchRequestApiModel.getSearchString()).getTextCriteriaMatchPhrase();
```

Comment:
You could merge all these calls into one sicCodeSearchRequestApiModel.getSearchString() on line 18, assign to short var name and make the code more readable.

Reply:
Any thoughts on this?
I was suggesting:
Line 18 String searchString = sicCodeSearchRequestApiModel.getSearchString();
Lines 19, 25, 28 use searchString.foo()

Reply:
Already done in the latest commit (two hours ago)